src/main/java/org/mbari/m3/dive/visualizer/DiveAnnotationService.java

```diff
@@ -121,121 +76,144 @@ private void getRovDiveAnnotations(ServerRequest request, ServerResponse respons
+        for (int i = 0; i < media.size(); i++) {
             String uri = media.get(i).getAsJsonObject().get("uri").toString();
-            if(uri.charAt(uri.length()-2)=='4'){
+            if (uri.charAt(uri.length() - 2) == '4') {
                 allDiveVideos.add(media.get(i).getAsJsonObject().get("uri"));
             }
         }
         return allDiveVideos;
     }
 
-    private JsonObject getVideoLinksAndAnnotations(JsonObject allAnnotationData){
-        if(allAnnotationData.isJsonNull()) {
-            System.out.println("EMPTY ANNOTATION TREE - AnnotationServieHelper.linksAndAnnotations()");
+
+    public JsonObject getVidsAndAnnotations(JsonObject allAnnotationData) {
+        if (allAnnotationData.isJsonNull()) {
+            log.log(Level.WARNING, "Annotation Data empty - DiveAnnotationService.getVideoLinksAndAnnotations()");
             return null;
         }
         JsonArray allMedia = getMedia(allAnnotationData);
         JsonArray videoLinks = getVideoLinks(allAnnotationData);
-        JsonObject linksAndUUID = new JsonObject(); //  linksAndUUID eventually become links and their annotations 
+        JsonObject linksAndUUID = new JsonObject(); // linksAndUUID eventually become links and their annotations
+        JsonArray mp4WithNoAvailableMov = new JsonArray();
 
         for(int j = 0; j < videoLinks.size();j++) {
             for(int i = 0; i < allMedia.size(); i++) {   
                 String video_reference_uuid = "";
                 if(videoLinks.get(j).toString().equals(allMedia.get(i).getAsJsonObject().get("uri").toString())){
+                    
+                    String uri = videoLinks.get(j).getAsString();
+                    linksAndUUID.add(videoLinks.get(j).getAsString(), new JsonObject());
+                    //linksAndUUID.add(videoLinks.get(j).toString(), new JsonObject());
                     video_reference_uuid = getVideoReferenceUUID(allMedia.get(i).getAsJsonObject().get("video_uuid").toString(),allAnnotationData);
-                    // length == 0 means that this mp4 video does not have a matching mov file. 
-                    // It needs a matching mov file for us to get the
-                    // video_reference_uuid that will get us the matching annotations
+                                                            // length == 0 means that this mp4 video does not have a matching mov file. 
+                                                            // It needs a matching mov file for us to get the
+                                                            // video_reference_uuid that will get us the matching annotations
                     if(video_reference_uuid.length()==0){                                           
-                        linksAndUUID.addProperty(videoLinks.get(j).toString(), "No video_reference_uuid");
+                        linksAndUUID.get(uri).getAsJsonObject().addProperty("video_reference_uuid", "No video_reference_uuid");
```

Comment:
Maybe break this into a separate line? 
There's a few long lines like this, you can break this up as each time you use a method break it up into a separate line, or leave fetching the object on one line, and break up changing the object into separate lines.
```suggestion
                        linksAndUUID.get(uri).getAsJsonObject()
                            .addProperty("video_reference_uuid", "No video_reference_uuid");
```